feature_engine/encoding/similarity_encoder.py

```diff
@@ -226,9 +249,12 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
                     X[var]
                     .astype(str)
                     .value_counts(dropna=True)
+                    .drop("nan", errors="ignore")
                     .head(self.top_categories)
                     .index.tolist()
                 )
+        if self.keywords:
```

Comment:
If we update the dictionary here, and the dictionary keys contain all the variables in `variables_`, we are running all of the logic between 226 and this line, for no reason. Because those keywords will not be used. 

Would it not be better to check, at around line 223 that:

- if, all of `variables_` are in `keywords.keys()`, then just copy dict. 
- If some of `variables_` are in keywords.keys(), then apply the logic to find keywords in the remaining variables
- If keywords is None, then apply the logic in all `variables_`

What do you think about this?

Reply:
No, it will be worse. It will make logic more complicated (which in turn will mess up coverage and tests) and won't give any speed up. 

This way it is straightforward and readable.